src/components/PeoplePage.tsx

```diff
@@ -1,32 +1,109 @@
+import React, { useEffect, useState } from 'react';
+import { useSearchParams } from 'react-router-dom';
 import { PeopleFilters } from './PeopleFilters';
 import { Loader } from './Loader';
 import { PeopleTable } from './PeopleTable';
+import { getPeople } from '../api';
+import { Person } from '../types';
+
+export const PeoplePage: React.FC<{}> = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState(false);
+  const [searchParams] = useSearchParams();
+
+  let visiblePeople = [...people];
+
+  const sex = searchParams.get('sex') || null;
+  const query = searchParams.get('query') || null;
+  const centuries = searchParams.getAll('centuries') || [];
+  const sorting = searchParams.get('sort') || null;
+  const order = searchParams.get('order') || null;
+
+  useEffect(() => {
+    const loadPeople = async () => {
+      try {
+        setPeople(await getPeople());
+      } catch (err) {
+        setError(true);
+      } finally {
+        setIsLoading(false);
+      }
+    };
+
+    loadPeople();
+  }, []);
+
+  const isErrorMessageVisible = !isLoading && error;
+  const isPeopleTableVisible = !isLoading && !error && people.length > 0;
+
+  if (sex) {
+    visiblePeople = visiblePeople.filter(person => person.sex === sex);
+  }
+
+  if (query) {
+    visiblePeople = visiblePeople
+      .filter(person => {
+        const searchFilter = `${person.motherName?.toLowerCase()}${person.fatherName?.toLowerCase()}${person.name.toLowerCase()}`;
```

Comment:
It would be cool to use destructuring in such places:
```
.filter({ motherName, fatherName, name }=> {
  const searchFilter = `${motherName?.toLowerCase()}${fatherName?.toLowerCase()}${name.toLowerCase()}`;
  ...
```